flink-cdc-common/src/main/java/com/ververica/cdc/common/event/TableId.java

```diff
@@ -0,0 +1,96 @@
+/*
+ * Copyright 2023 Ververica Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.ververica.cdc.common.event;
+
+import javax.annotation.Nullable;
+
+import java.util.Objects;
+
+/** Unique identifier for a database table. */
+public class TableId {
+
+    public static final String DEFAULT_NAMESPACE = "default";
+
+    private final String namespace;
+    private final String schemaName;
+    private final String tableName;
+
+    public TableId(String namespace, String schemaName, String tableName) {
+        this.namespace = Objects.requireNonNull(namespace);
+        this.schemaName = Objects.requireNonNull(schemaName);
+        this.tableName = Objects.requireNonNull(tableName);
+    }
+
+    public static TableId of(String namespace, String schemaName, String tableName) {
+        return new TableId(namespace, schemaName, tableName);
+    }
+
+    public static TableId of(String schemaName, String tableName) {
+        return new TableId(DEFAULT_NAMESPACE, schemaName, tableName);
+    }
+
+    public static TableId parse(String tableId) {
+        String[] parts = Objects.requireNonNull(tableId).split("\\.");
+        if (parts.length == 3) {
+            return of(parts[0], parts[1], parts[2]);
+        } else if (parts.length == 2) {
+            return of(parts[0], parts[1]);
+        }
+        throw new IllegalArgumentException("Invalid tableId: " + tableId);
+    }
+
+    public String identifier() {
+        return namespace + "." + schemaName + "." + tableName;
+    }
+
+    @Nullable
```

Comment:
In the constructor, it is required that the namespace is not null. Why the "@Nullable" is used here?

Reply:
Thanks for catching this.